x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -324,8 +326,10 @@ public void executePlan(
             listener.onFailure(new IllegalStateException("expected data node plan starts with an ExchangeSink; got " + dataNodePlan));
             return;
         }
-        Map<String, OriginalIndices> clusterToConcreteIndices = transportService.getRemoteClusterService()
-            .groupIndices(SearchRequest.DEFAULT_INDICES_OPTIONS, PlannerUtils.planConcreteIndices(physicalPlan).toArray(String[]::new));
+
+        // Returns the concrete/resolved indices used in the FROM command of the query.
+        Map<String, OriginalIndices> clusterToConcreteIndices = getIndices(execInfo, EsqlExecutionInfo.Cluster::getConcreteIndices);
```

Comment:
Grouping concrete indices should be possible, but it is preferable to store result in EsqlExecutionInfo instead of traversing plan and grouping indices again (especially if we have a pattern resolved to hundreds of indices).

Reply:
This is actually not correct in case of subqueries.
For example with `FROM index-1, (FROM index-2)` EsqlExecutionInfo would contain both `index-1,index-2` while nested plan should only query `index-2`. I should find another way of handling this. Probably I need to keep extra information in the plan/EsRelation itself.